### PR TITLE
Enabling editing a node

### PR DESCRIPTION
**Pull request by dawehner**
https://github.com/jsdrupal/drupal-admin-ui/issues/251

Current TODOs:

* [x] Figure out how to query for all referenced entities when fetching the content
* [x] Fix the file upload to include the referenced file
* [x] Ensure all content actually gets prepoulated: categories, authors, tags, images, authored by are all empty right now
* [ ] Posting fails right now because it sends the status field which is not allowed to be sent due to workflow moderation. Ideally the PATCH would just POST changed fields
* [ ] Ensure saving actually works

### CODE REVIEW

File: src/routes.js [hunk 1]
```diff
@@ -15,7 +16,15 @@ const routes = {
   '/admin/reports/dblog': Dblog,
   '/node/add': AddContent,
   '/node/add/recipe': () => (
-    <NodeForm entityTypeId="node" bundle="recipe" widgets={widgets} />
+    <NodeAddForm entityTypeId="node" bundle="recipe" widgets={widgets} />
+  ),
+  '/node/:bundle/:id/edit': ({ match }) => (
```
Comment (dawehner) [inline src/routes.js]: Note: We need to pass along the bundle as there is not way to load a single piece of content without knowing the bundle.

File: src/actions/content.js [hunk 2]
```diff
@@ -107,6 +107,35 @@ function* loadContent(action) {
...
+});
+
+export const CONTENT_SINGLE_LOADED = 'CONTENT_SINGLE_LOADED';
+function* loadSingleContent(action) {
+  const { payload: { bundle, id } } = action;
+  try {
+    yield put(resetLoading());
+    yield put(showLoading());
+
+    const content = yield call(api, 'content_single', { parameters: { bundle, id }, queryString: {
```
Comment (dawehner) [inline src/actions/content.js]: Should we first look up in state.content.contentById whether the content already exists?

File: src/components/02_atoms/Widgets/FileUploadWidget.js [hunk 3]
```diff
@@ -50,153 +51,220 @@ const styles = {
...
+      } else {
+        ids = [this.props.value.data.id];
+      }
+      this.fetchEntitites(entityTypeId, bundle, ids).then(({ data: items }) => {
+        this.setState({
+          selectedItems: items.map(({ id, attributes }, index) => ({
+            id,
+            type: 'file--file',
+            [entityTypeId]: attributes,
+            meta: this.props.value.data[index].meta,
```
Comment (mattgrill) [inline src/components/02_atoms/Widgets/FileUploadWidget.js]: @dawehner Here's an interesting issue. If there is only one entity returned, you can't fetch it by it's index; it's at `this.props.value.data.meta`. You can probably check like, `Array.isArray(this.props.value.data) ? this.props.value.data[index].meta : this.props.value.data.meta`

Comment (dawehner) [reply src/components/02_atoms/Widgets/FileUploadWidget.js]: good point!

**Comment by mattgrill** [comment]
Alright. 🎉👍 Great job! I'm really excited to get this in, and give us some more time to address the issues in a series of followup issues.

**Pull request merged**